src/DropDown/DropDown.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[];
+  onSelect?: (person: Person | null) => void;
+  delay: number
+};
+
+export const DropDown: React.FC<Props> = React.memo(({
+  people,
+  onSelect = () => { },
+  delay,
+}) => {
+  const [query, setQuery] = useState('');
+  const [apliedQuery, setApliedQuery] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+
+  const applyQuery = debounce(setApliedQuery, delay);
+
+  const visiblePeople = useMemo(() => (people.filter(person => {
+    const normalizedQuery = apliedQuery.trim().toLowerCase();
+
+    return person.name.trim().toLowerCase().includes(normalizedQuery);
+  })), [apliedQuery, people]);
+
+  const resultToShow = (visiblePeople.length > 0);
+
+  const isPeopleListShown = resultToShow && isFocused;
+
+  const queryHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const selectHander = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    person: Person,
+  ) => {
+    event.preventDefault();
+    onSelect(person);
+    setIsFocused(false);
+    setQuery(person.name);
+  };
+
+  const resetHandler = () => {
```

Comment:
this does not clear the search query correct

steps to reproduce:
input 'car' - see dropdown changes
use the reset button in the input - input is cleared
now click into the input to show the dropdown - you see dropdown filtered on "car" with empty input

Reply:
That happens because you have `selectedPeople` memoized, but their dependencies are not changed in scenario above